src/components/MarketSelector/PoolSelector.tsx

```diff
@@ -54,36 +57,59 @@ export function PoolSelector({
   showBalances,
   onSelectMarket,
   getMarketState,
+  showAllPools = false,
+  showIndexIcon = false,
 }: Props) {
   const [isModalVisible, setIsModalVisible] = useState(false);
   const [searchKeyword, setSearchKeyword] = useState("");
 
   const marketsOptions: MarketOption[] = useMemo(() => {
-    return markets
-      .filter((market) => !market.isDisabled && getMarketIndexName(market) === selectedIndexName)
+    const allMarkets = markets
+      .filter((market) => !market.isDisabled && (showAllPools || getMarketIndexName(market) === selectedIndexName))
       .map((marketInfo) => {
+        const indexName = getMarketIndexName(marketInfo);
         const poolName = getMarketPoolName(marketInfo);
         const marketToken = getByKey(marketTokensData, marketInfo.marketTokenAddress);
         const gmBalance = marketToken?.balance;
         const gmBalanceUsd = convertToUsd(marketToken?.balance, marketToken?.decimals, marketToken?.prices.minPrice);
         const state = getMarketState?.(marketInfo);
 
         return {
+          indexName,
           poolName,
+          name: marketInfo.name,
           marketInfo,
           balance: gmBalance || BigNumber.from(0),
           balanceUsd: gmBalanceUsd || BigNumber.from(0),
           state,
         };
       });
-  }, [getMarketState, marketTokensData, markets, selectedIndexName]);
+    const marketsWithBalance: MarketOption[] = [];
+    const marketsWithoutBalance: MarketOption[] = [];
+
+    for (let market of allMarkets) {
+      if (market.balance.gt(0)) {
+        marketsWithBalance.push(market);
+      } else {
+        marketsWithoutBalance.push(market);
+      }
+    }
+
+    const sortedMartketsWithBalance = marketsWithBalance.sort((a, b) => {
+      return b.balanceUsd?.sub(a.balanceUsd || 0).gt(0) ? 1 : -1;
+    });
+
+    return [...sortedMartketsWithBalance, ...marketsWithoutBalance];
+  }, [getMarketState, marketTokensData, markets, selectedIndexName, showAllPools]);
 
   const marketInfo = marketsOptions.find(
     (option) => option.marketInfo.marketTokenAddress === selectedMarketAddress
   )?.marketInfo;
 
   const filteredOptions = marketsOptions.filter((option) => {
-    return option.poolName.toLowerCase().indexOf(searchKeyword.toLowerCase()) > -1;
+    const name = option.name.toLowerCase();
+    const search = searchKeyword.toLowerCase();
```

Comment:
just i thing to consider for future: may be better to define such vars outside of the loop since it just does the same each time (`searchKeyword` is the same for each iteration)